corehq/apps/userreports/models.py

```diff
@@ -146,6 +146,29 @@ class DataSourceActionLog(models.Model):
     skip_destructive = models.BooleanField(default=False)
 
 
+class DataSourceRowTransactionLog(models.Model):
+    CREATE = "create"
+    UPDATE = "update"
+    DELETE = "delete"
+
+    domain = models.CharField(max_length=126, null=False, db_index=True)
+    date_created = models.DateTimeField(auto_now_add=True)
+    action = models.CharField(max_length=32, choices=(
+        (CREATE, CREATE),
+        (UPDATE, UPDATE),
+        (DELETE, DELETE),
+    ), null=False, blank=False)
+    data_source_id = models.UUIDField(null=False, blank=False)
```

Comment:
Hmm, I think we might want to index this column as well

Reply:
Ah, right